openapi3gen/openapi3gen.go

```diff
@@ -355,9 +397,40 @@ func (g *Generator) generateWithoutSaving(parents []*theTypeInfo, t reflect.Type
 		}
 	}
 
+	if !g.opts.exportComponentSchemas.ExportComponentSchemas || t.Kind() != reflect.Struct {
+		return openapi3.NewSchemaRef(t.Name(), schema), nil
+	}
+
+	// Best way I could find to check that
+	// this current type is a generic
+	isGeneric, err := regexp.Match(`^.*\[.*\]$`, []byte(t.Name()))
```

Comment:
How about just checking that the last rune is `]`?

Reply:
would be an array in that case?

Reply:
ah yes, then `runes[-1] == ']' && runes[-2] != '['`